src/test/java/com/orzechazo/eshop/services/UserServiceImplIT.java

```diff
@@ -0,0 +1,142 @@
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private OrderRepository orderRepository;
+    private UserServiceImpl userService;
+    private final static String DB_USER_LOGIN = BootstrapUsersAndOrders.DB_USER_LOGIN;
+    private int DEFAULT_DB_USER_COUNT;
+    private BootstrapUsersAndOrders bootstrap;
+    @BeforeEach
+    void setUp() {
+        bootstrap = new BootstrapUsersAndOrders(orderRepository, userRepository);
+        bootstrap.loadData();
+        DEFAULT_DB_USER_COUNT = bootstrap.getUsers().size();
+
+        userService = new UserServiceImpl(userRepository);
+    }
+
+    @Test
+    void getAllUsers() {
+        List<String> dbUsersLoginList = bootstrap.getUsers().stream()
+                .map(User::getLogin).toList();
+        //when
+        List<UserDto> userDtos = userService.getAllUsers();
+        List<String> userDtosLogins = userDtos.stream().map(UserDto::getLogin).toList();
+        //then
+        assertThat(dbUsersLoginList).containsExactlyInAnyOrderElementsOf(userDtosLogins);
+    }
+
+    @Test
+    void getUserByLogin() {
+        UserDto returnedDto = userService.getUserDtoByLogin(DB_USER_LOGIN);
+
+        assertEquals("user1",returnedDto.getLogin());
+        assertNull(returnedDto.getPassword());
+    }
+
+    @Test
+    void getUserByLoginNotExistingInDatabase() {
+        Exception exception = assertThrows(ResourceNotFoundException.class,
+                () -> userService.getUserDtoByLogin("notExistingUser"));
+        assertEquals("User: notExistingUser doesn't exist in database.",exception.getMessage());
+    }
+
+    @Test
+    void createUser() {
+        UserDto newUser = UserDto.builder().login("newUser")
+                .password("testPassword").build();
+        //when
+        UserDto createdDto = userService.createUser(newUser);
+        //then
+        assertEquals("newUser",createdDto.getLogin());
+        assertNull(createdDto.getPassword());
+        assertEquals(DEFAULT_DB_USER_COUNT + 1,userRepository.count());
+    }
+
+    @Test
+    void testTryToCreateUserWhoseLoginIsAlreadyInDatabase() {
+        UserDto userDto = UserDto.builder().login(DB_USER_LOGIN).password("testPassword").build();
+
+        Exception exception = assertThrows(BadRequestException.class,
+                () -> userService.createUser(userDto));
+        assertEquals("User: " + DB_USER_LOGIN + " is already in database.",exception.getMessage());
+    }
+
+    @Test
+    void testTryToCreateTheSameUserTwice() {
+        UserDto userDto = UserDto.builder().login("newUser").build();
+
+        userService.createUser(userDto);
+        Exception exception = assertThrows(BadRequestException.class,
+                () -> userService.createUser(userDto));
+        assertEquals("User: newUser is already in database.",exception.getMessage());
+    }
+
+    @Test
+    void updateUser() {
+        UserDto userToUpdate = UserDto.builder().login(DB_USER_LOGIN).password("newPassword").build();
```

Comment:
This test is not proving anything, we might update new password here but you cannot check it as it is not returned in dto, is there any other field that can be updated and checked?

Reply:
At the moment there is no other field that we can update and check the way the method is written, I guess I will rewrite the method to check for other fields aswell and then correct the test